packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/tasks/MdlToStockflowResponseHandler.java

```diff
@@ -34,7 +35,13 @@ public static class Response {
 	public TaskResponse onSuccess(final TaskResponse resp) {
 		try {
 			final Response modelResp = objectMapper.readValue(resp.getOutput(), Response.class);
-			final Model model = modelService.createAsset(modelResp.getResponse());
+			Model model = modelResp.getResponse();
+			final ConversionAdditionalProperties props = resp
+					.getAdditionalProperties(ConversionAdditionalProperties.class);
+			// override the default stockflow name / description
+			model.getHeader().setName(props.getFileName());
+			model.getHeader().setDescription(props.getFileName());
+			model = modelService.createAsset(model);
```

Comment:
We shoulsd also set the `TerariumAsset.name`
Also are you sure the `getFileName` is the best name, should we read the name from the json content?
```suggestion
			model.getHeader().setDescription(props.getFileName());
			model.setName(props.getFileName());
			model = modelService.createAsset(model);
```

Reply:
fixed. I'm pretty sure that the problem is the json content. It defaults to "SIR model"